taskcat/s3_stage.py

```diff
@@ -0,0 +1,149 @@
+
+class S3APIResponse:
+    def __init__(self, x):
+        self._http_code = x['ResponseMetadata']['HTTPStatusCode']
+
+    @property
+    def ok(self):
+        if self._http_code == 200:
+            return True
+        return False
+
+class S3BucketCreatorException(Exception):
+    pass
+
+class S3BucketCreator:
+    def __init__(self, config: Config):
+        self.name = ""
+        self.public = False
+        self.tags = []
+        self.region = 'us-east-1'
+        self.sigv4 = True
+        self._config = config
+        self._c = None
+
+    @property
+    def acl(self):
+        return self._acl
+
+    @property
+    def policy(self):
+        return self._policy
+
+    def _create_in_region(self, region):
+        if region == 'us-east-1':
+            response = self._c.create_bucket(
+                ACL=self.acl,
+                Bucket=self.name
+            )
+        else:
+            response = self._c.create_bucket(
+                ACL=self.acl,
+                Bucket=self.name,
+                CreateBucketConfiguration={
+                    'LocationConstraint': region
+                }
+            )
+
+        return S3APIResponse(response)
+
+    def _return_sigv4_policy(self, bucket):
+        policy = F"""{
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "Test",
+                    "Effect": "Deny",
+                    "Principal": "*",
+                    "Action": "s3:*",
+                    "Resource": "arn:aws:s3:::{bucket}/*",
+                    "Condition": {
+                         "StringEquals": {
+                               "s3:signatureversion": "AWS"
+                         }
+                    }
+                }
+            ]
+        }"""
+
+    def create(self):
```

Comment:
this method is a bit long, can we split it up, maybe create a `_assert_bucket_exists` for line 78-84 and `_create` for 91-104